tests/TestHarness/Node.py

```diff
@@ -43,145 +34,56 @@ def __init__(self, host, port, nodeId, pid=None, cmd=None, walletMgr=None, nodeo
         self.nodeId=nodeId
         if Utils.Debug: Utils.Print("new Node host=%s, port=%s, pid=%s, cmd=%s" % (self.host, self.port, self.pid, self.cmd))
         self.killed=False # marks node as killed
-        self.endpointHttp="http://%s:%d" % (self.host, self.port)
-        self.endpointArgs="--url %s" % (self.endpointHttp)
+
         self.infoValid=None
         self.lastRetrievedHeadBlockNum=None
         self.lastRetrievedLIB=None
         self.lastRetrievedHeadBlockProducer=""
         self.transCache={}
-        self.walletMgr=walletMgr
         self.missingTransaction=False
         self.popenProc=None           # initial process is started by launcher, this will only be set on relaunch
         self.lastTrackedTransactionId=None
-        self.nodeosVers=nodeosVers
-        if self.nodeosVers=="v2":
+        self.pollVersion()
+
+    def pollVersion(self):
+        info = self.getInfo(silentErrors=True)
+        timeout = datetime.now() + timedelta(seconds=20)
+        while info == None:
+            time.sleep(0.5)
+            info = self.getInfo(silentErrors=True)
+            if datetime.now() > timeout:
+                break
+        else:
+            self.configureVersion(info)
+
+    def configureVersion(self, info):
+        self.nodeosVers=info["server_version_string"]
+        if 'v2' in self.nodeosVers:
```

Comment:
We should probably have some exceptional or graceful handling of the timeout scenario so that either we have a way to handle in info is still None and just default to the else in configureVersion or else raise an exception from pollVersion instead of waiting to try to access info in configureVersion.  Failing that, maybe the dynamic version discovery should be reverted until a holistic approach can be implemented here and just use the original static passed in version number.

Reply:
Largely addressed in https://github.com/AntelopeIO/leap/pull/192/commits/c5e2bcc983ba09bb0745993301d44923a2e9c7fa with final cleanup in https://github.com/AntelopeIO/leap/pull/192/commits/094448ac068881d5557ccea8900c20edea5bc4a0